### PR TITLE
Added get_blobs_list method to WasbHook

### PR DESCRIPTION
**Pull request by jayrumi**
The function returns list with blobs. Also it provides possibility to pass on any parameters for list_blobs() function using **kwargs

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: airflow/providers/microsoft/azure/hooks/wasb.py [hunk 1]
```diff
@@ -88,6 +88,23 @@ def check_for_prefix(self, container_name, prefix, **kwargs):
                                              num_results=1, **kwargs)
         return len(list(matches)) > 0
 
+    def get_blobs_list(self, container_name, prefix, **kwargs):
```
Comment (turbaszek) [inline airflow/providers/microsoft/azure/hooks/wasb.py]:
```suggestion
    def get_blobs_list(self, container_name: str, prefix: str, **kwargs):
```

**Comment by turbaszek** [comment]
@jayrumi would you mind extending the tests for this method? It should be a simple one but having test helps to avoid regression

**Comment by jayrumi** [comment]
@turbaszek I don't mind. As I understand the test will be provided by you, won't it?

**Comment by turbaszek** [comment]
@jayrumi please add those test :)

**Comment by jayrumi** [comment]
@turbaszek could you help me with it or maybe provide some links how to do it? It's my first experience in contributing and I'm not a mature Python developer ) Just learning

**Comment by turbaszek** [comment]
Sure no, problem!
Here are existing tests for WasbHook:
https://github.com/apache/airflow/blob/master/tests/providers/microsoft/azure/hooks/test_wasb.py

and that is the place where you should add your test. Now, let's consider this existing test:
https://github.com/apache/airflow/blob/ef98edf4da2d9b74d5cf5b21e81577b3151edb79/tests/providers/microsoft/azure/hooks/test_wasb.py#L119-L127

First of all we are using [mock library](https://docs.python.org/3/library/unittest.mock.html) to mock the behaviour of `BlockBlobService`. By doing this we assume that `BlockBlobService` works right (it's Azure stuff not ours) and all we have to check is that we use it right. By that it mostly means that we properly pass arguments.

In this line:
https://github.com/apache/airflow/blob/ef98edf4da2d9b74d5cf5b21e81577b3151edb79/tests/providers/microsoft/azure/hooks/test_wasb.py#L122
we create mocked instance of `BlockBlobService` (think about `return_value` as a result of calling `BlockBlobService()`). Next we create hook instance and we call the method:
https://github.com/apache/airflow/blob/ef98edf4da2d9b74d5cf5b21e81577b3151edb79/tests/providers/microsoft/azure/hooks/test_wasb.py#L123-L124
No magic here, just simple invocation 👌 

Then we call this magic thing:
https://github.com/apache/airflow/blob/ef98edf4da2d9b74d5cf5b21e81577b3151edb79/tests/providers/microsoft/azure/hooks/test_wasb.py#L125-L127

In this way we asserts that `BlockBlobService().get_blob_to_path` method was called with those arguments: `'container', 'blob', 'path', max_connections=1`. As you can see, those are same as the one used in ` hook.get_file('path', 'container', 'blob', max_connections=1)`

And that's it, we checked that arguments of `WasbHook.get_file` are properly passed to `BlockBlobService.get_blob_to_path`.

You have to create similar test. I encourage you to take a look at the mock package to better understand how it works - it will help you in future 💪 Feel free to ask in case of any questions 👌

**Comment by jayrumi** [comment]
@turbaszek you are amazing man! Thank you for help. I'll add a test near future. 
And what's about CI Build failed tests? What should I do?

**Comment by turbaszek** [comment]
> @turbaszek you are amazing man!

Thanks! 

> And what's about CI Build failed tests? What should I do?

The quarantined are flaky and fail from time to time so no worry here, we know that. In case of static checks I recommend taking a look at https://github.com/apache/airflow/blob/master/STATIC_CODE_CHECKS.rst However if you have no time, take a look the failed build and you will see that there's a redundant space - remove it and you should be good. But I honestly recommend taking a look at the linked page and our pre-commits ;)

**Review by turbaszek — APPROVED** [review]

File: airflow/providers/microsoft/azure/hooks/wasb.py [hunk 2]
```diff
@@ -88,6 +88,23 @@ def check_for_prefix(self, container_name, prefix, **kwargs):
                                              num_results=1, **kwargs)
         return len(list(matches)) > 0
 
+    def get_blobs_list(self, container_name: str, prefix: str, **kwargs):
+        """
+        Return a list of blobs from path defined in prefix param
```
Comment (turbaszek) [inline airflow/providers/microsoft/azure/hooks/wasb.py]:
```suggestion
        Return a list of blobs from path defined in prefix param
```

File: airflow/providers/microsoft/azure/hooks/wasb.py [hunk 3]
```diff
@@ -88,6 +88,23 @@ def check_for_prefix(self, container_name, prefix, **kwargs):
...
+        :type prefix: str
+        :param kwargs: Optional keyword arguments that
+            `BlockBlobService.list_blobs()` takes (num_results, include,
+            delimiter, marker, timeout)
+        :type kwargs: object
+        :return: List of blobs.
+        :rtype: list(azure.storage.common.models.ListGenerator)
+        """
+        return self.connection.list_blobs(container_name, prefix, **kwargs)
+
```
Comment (turbaszek) [inline airflow/providers/microsoft/azure/hooks/wasb.py]:
```suggestion

```

File: airflow/providers/microsoft/azure/hooks/wasb.py [hunk 4]
```diff
@@ -88,6 +88,22 @@ def check_for_prefix(self, container_name, prefix, **kwargs):
...
+        :param prefix: Prefix of the blob.
+        :type prefix: str
+        :param kwargs: Optional keyword arguments that
+            `BlockBlobService.list_blobs()` takes (num_results, include,
+            delimiter, marker, timeout)
+        :type kwargs: object
+        :return: List of blobs.
+        :rtype: list(azure.storage.common.models.ListGenerator)
+        """
+        return self.connection.list_blobs(container_name, prefix, **kwargs)
```
Comment (turbaszek) [inline airflow/providers/microsoft/azure/hooks/wasb.py:106]:
```suggestion
        return self.connection.list_blobs(container_name, prefix, **kwargs)

```

**Comment by turbaszek** [comment]
Thanks @jayrumi !

**Comment by jayrumi** [comment]
@turbaszek thank you for awesome help too!

**Pull request merged**